### ISSUE TITLE
Is it possible to get local domains working from the host when running Lando in WSL?

### ISSUE THREAD

**Issue by k1sul1** [lando/lando#2598]
I've used Linux for all of my development for the past 5 years, but I guess I'm getting old and grumpy because I'm getting real tired of doing everything myself, so I'm giving Windows + WSL a chance. So far I've tried multiple local environments, and most had the same problem; they didn't run in WSL, and fs performance from WSL to /mnt is terrible, so things like `git status` took forever.

I got Lando working inside WSL with the help of this blog post; https://blog.calevans.com/2020/06/18/making-lando-work-inside-wsl2/

Like the author, I too dabble in the dark art of WordPress most of the time, so I set up a WordPress instance as my hello world: 

```
❯ lando info
[ { service: 'appserver',
    urls:
     [ 'https://localhost:32785',
       'http://localhost:32786',
       'http://my-first-wordpress-app.lndo.site/',
       'https://my-first-wordpress-app.lndo.site/' ],
    type: 'php',
    healthy: true,
    via: 'apache',
    webroot: 'wordpress',
    config: { php: '/home/k1sul1/.lando/config/wordpress/php.ini' },
    version: '7.3',
    meUser: 'www-data',
    hasCerts: true,
    hostnames: [ 'appserver.myfirstwordpressapp.internal' ] },
  { service: 'database',
    urls: [],
    type: 'mysql',
    healthy: true,
    internal_connection: { host: 'database', port: '3306' },
    external_connection: { host: '127.0.0.1', port: '32787' },
    healthcheck: 'bash -c "[ -f /bitnami/mysql/.mysql_initialized ]"',
    creds: { database: 'wordpress', password: 'wordpress', user: 'wordpress' },
    config: {},
    version: '5.7',
    meUser: 'www-data',
    hasCerts: false,
    hostnames: [ 'database.myfirstwordpressapp.internal' ] } ]
```

Here's the thing. The localhost urls work fine, and the site works. However, the my-first-wordpress-app.lndo.site urls do not work, it simply refuses to connect when I try to access it from the browser. At first I thought I simply require a hosts entry on the host.

I use [wsl2host](https://github.com/shayne/go-wsl2-host) to sync the WSL ip address with all the hosts that I use, so the end of my hostsfile already looks like this:

172.21.187.253 my-first-wordpress-app.lndo.site    # alias: Ubuntu-20.04; managed by wsl2-host

But no cigar. I tried adding `127.0.0.1 my-first-wordpress-app.lndo.site` to the WSL hostsfile, but that didn't help either. 

For the time being, I can manage with the localhost URLs. WordPress internals are "not very nice"; there's a severe limitation with WordPress Network, it can only run on ports 80 and 443. I don't need to work on any multisites (Network) at the moment, but I do have some that may require work in the future, and I'd like to be able to work on those without having to hack something together on a moments notice.

**Comment by k1sul1** [comment]
To answer my own qvimuestion, seems so. 

I was able to get wordpress.lndo.site domains working by following this blog post: https://roots.io/guides/dockerize-local-bedrock-and-sage-development-with-lando/

I excluded the bedrock.test proxy to reduce the amount of moving parts and _it just works_.

**Comment by dustinleblanc** [comment]
This shouldn't be necessary, using Windows 10 and WSL2, I am able to get Lando proxy urls to work out of the box with no changes. My guess is that something is broken with DNS resolution on the host system

**Comment by k1sul1** [comment]
Yes, it seems the hosts modifications are unnecessary, and I did say that I got it working.

The WordPress side at least. I have a webpack-dev-server in one of the themes and I haven't been able to expose it to the host yet. It's bound at 0.0.0.0, so it should work, but it doesn't. Minor annoyance, the build works _but it would be nice_ to get HMR too. 

```
name: sitename
recipe: wordpress
proxy:
  node: # webpack-dev-server url
    - localhost:8080
    # - hotapp.lndo.site
config:
  php: "7.3"
  via: nginx
  webroot: web
  database: mariadb
  xdebug: true

# More containers such as build processes or CI flows
services:
  node:
    type: node
    services:
      ports:
        - "8080:8080"

# Example usage: run `lando npm run build` or `lando yarn run build`
tooling:
  npm:
    service: node
  yarn:
    service: node
```

```
// webpack.config.babel.js
// ...
    devServer: {
      historyApiFallback: true,
      host: "0.0.0.0",
      port: 8080,
      open: false,
      hotOnly: true,
      clientLogLevel: "none",
      disableHostCheck: true, // unsafe but important

      proxy: {
        "/": {
          target: "https://sitename.lndo.site/",
          changeOrigin: true,
          autoRewrite: true,
          secure: false,
          headers: {
            "X-Proxy": "webpack-dev-server",
          },
        },
      },
// ...
}
```

**Comment by Xilonz** [comment]
I use the following setup \w laravel mix, but the browsersync options should be the same. It exposes 8080 internally so you can add that to
```
mix.setPublicPath('./dist')
   .browserSync({
        proxy: 'http://appserver_nginx',
        https: false,
        open: false,
        socket: {
            domain: 'https://theme-sitename.lndo.site',
            port: 80
        }
   });
```
in lando.yml:
```
proxy:
  node:
    - theme-sitename.lndo.site:3000
```  

Check out https://docs.lando.dev/guides/how-to-wire-up-browsersync-and-gulp-in-my-lando-app.html#configuring-gulp-to-push-scss

**Comment by DrupalPhil** [comment]
I've followed the instructions [here](https://docs.lando.dev/guides/how-to-wire-up-browsersync-and-gulp-in-my-lando-app.html#configuring-gulp-to-push-scss) but for some reason I'm only getting `Bad Gateway` when I get here:

> Visit your browsersync url `https://bs.myproject.lndo.site` in your browser. It should load normally.

I've gone through the instructions a number of times, it's a vanilla install, and there isn't any fancy footwork besides this.

I've review the latest release notes and nothing suggests this shouldn't still work.

Is this issue exclusive to me or are others having the same issue?

I've tried with Lando v3.0.14 & v3.0.26
